src/Xamarin.Android.Build.Tasks/Tasks/GenerateJavaStubs.cs

```diff
@@ -290,16 +295,62 @@ void WriteTypeMappings (List<TypeDefinition> types)
 			using (var gen = UseSharedRuntime
 				? new TypeNameMapGenerator (types, Log.LogDebugMessage)
 			        : new TypeNameMapGenerator (ResolvedAssemblies.Select (p => p.ItemSpec), Log.LogDebugMessage)) {
-				UpdateWhenChanged (Path.Combine (OutputDirectory, "typemap.jm"), gen.WriteJavaToManaged);
-				UpdateWhenChanged (Path.Combine (OutputDirectory, "typemap.mj"), gen.WriteManagedToJava);
+				UpdateWhenChanged (Path.Combine (OutputDirectory, "typemap.jm"), "jm", gen.WriteJavaToManaged);
+				UpdateWhenChanged (Path.Combine (OutputDirectory, "typemap.mj"), "mj", gen.WriteManagedToJava);
 			}
 		}
 
-		void UpdateWhenChanged (string path, Action<Stream> generator)
+		void UpdateWhenChanged (string path, string type, Action<Stream> generator)
 		{
-			using (var stream = new MemoryStream ()) {
-				generator (stream);
-				MonoAndroidHelper.CopyIfStreamChanged (stream, path);
+			string fastDevType = AndroidFastDeploymentType?.Trim () ?? String.Empty;
+			using (var ms = new MemoryStream ()) {
+				if (!EmbedAssemblies || String.Compare ("Assemblies:Dexes", fastDevType, StringComparison.OrdinalIgnoreCase) == 0) {
```

Comment:
@dellis1972 should this be a `Contains("Dexes")`? can you put `Assemblies:Dexes:Resources` or something?

Reply:
Currently its only `Assemblies:Dexes`. That said we should be using the `_InstantRunEnabled` property which is already defined in this file https://github.com/xamarin/xamarin-android/blob/4f088670f2418d6291bd6d4b41fea0dd722081fb/src/Xamarin.Android.Build.Tasks/Tasks/GenerateJavaStubs.cs#L56